js/symbol/glyph_source.js

```diff
@@ -82,15 +84,19 @@ class GlyphSource {
         if (!remaining) callback(undefined, glyphs, fontstack);
 
         const onRangeLoaded = (err, range, data) => {
-            if (!err) {
-                const stack = this.stacks[fontstack][range] = data.stacks[0];
-                for (let i = 0; i < missing[range].length; i++) {
-                    const glyphID = missing[range][i];
-                    const glyph = stack.glyphs[glyphID];
-                    const rect  = atlas.addGlyph(uid, fontstack, glyph, buffer);
-                    if (glyph) glyphs[glyphID] = new SimpleGlyph(glyph, rect, buffer);
-                }
+            if (err) {
+                this.fire('error', {error: err});
```

Comment:
This is exactly how I would start 👌 .

The next step is to have the error event be propagated up to `Style` (from where it'll be propagated up to `Map`). You can see [how we do this on `ImageSprite` here.](http://github.com/mapbox/mapbox-gl-js/blob/abc7d0f2fbbd3beb8d8ec9a9753450b527e5133e/js/style/image_sprite.js#L25-L26)